rclcpp_lifecycle/src/lifecycle_node_interface_impl.cpp

```diff
@@ -190,59 +190,69 @@ LifecycleNode::LifecycleNodeInterfaceImpl::init(bool enable_communication_interf
         nullptr);
     }
   }
+
+  change_state_hdl = std::make_shared<ChangeStateHandlerImpl>(
```

Comment:
Passes each of these internal members of `LifecycleNodeInterfaceImpl` over to `ChangeStateHandlerImpl` as `change_state` changes these members (e.g., updates `current_state_` & `state_machine_`). This has the upside of allowing us to have `change_state_handler_impl.{c,h}pp` pulled out of this file for an arguably cleaner and easier read of the `change_state` process. However, using/changing these members may not be apparent to developers of `LifecycleNodeInterfaceImpl`.

Possible alternatives:
1. have `ChangeStateHandlerImpl` be an internal class definition + `friend` within `LifecycleNodeInterfaceImpl` (note made a `friend` [here](https://github.com/tgroechel/rclcpp/pull/1/files#diff-ce5719df945b934b0941e217c03a66152eede2b0e4133d890004fac5a914e38cR148) even though it is not needed, added to signal to developers but probably worth getting rid of unless we take this approach)
2. have a more explicit naming scheme for these referenced variables in `ChangeStateHandlerImpl`

Reply:
worth noting `current_state_` is a reflection of the underlying `state_machine_` state. `current_state_` is returned within [get_current_state](https://github.com/tgroechel/rclcpp/blob/rolling/rclcpp_lifecycle/src/lifecycle_node_interface_impl.cpp#L344) which returns a `const State&`